### DISCUSSION TITLE
Multiple databases with MongoDB `import { PrismaClient } from 'path/to/generated/client'` stuck with 'exports' is not defined

### DISCUSSION THREAD

**Discussion by xenogew** [prisma/prisma#19838] [Q&A] [answered]
### Bug description

I create a project with SvelteKit and design to connect to multiple databases: 
[1] PostgreSQL for membership data and 
[2] MongoDB for social feature like messages.

The [1] with default setup of SvelteKit and Prisma, there's no issue and can work properly.
But when I follow the tutorial blog (https://dulanwirajith.medium.com/how-to-connect-multiple-databases-using-prisma-in-nest-js-e8908529ba38), I found error and app is broken, and found error messages.

```
Internal server error: exports is not defined
```

Notice: `exports` is export with 's' in the end

### How to reproduce

1. Setup SvelteKit project and following this blog https://joyofcode.xyz/sveltekit-authentication-using-cookies (change SQLite to PostgreSQL), with setup databases with Docker
2. Create a new prisma file for 2nd DB, `prisma-message/schema.prisma`
3. prisma generate => `pnpm exec prisma generate --schema=./prisma-message/schema.prisma` and found example code `import { PrismaClient } from './node_modules/@messages/prisma/client'`
4. Create `lib/database.ts` file, and import the 1st database with `import { PrismaClient as MemberPrisma } from '@prisma/client';`
5. (while running application with `pnpm run dev`), add `import { PrismaClient } from '../node_modules/@messages/prisma/client';`
6. Found error.

```bash
[vite] Error when evaluating SSR module /node_modules/@messages/prisma/client/index-browser.js:
|- ReferenceError: exports is not defined
    at eval (/Users/my-user/sandbox/sw-society/node_modules/@messages/prisma/client/index-browser.js:4:23)
    at instantiateModule (file:///Users/my-user/sandbox/sw-society/node_modules/.pnpm/vite@4.3.9_@types+node@20.3.1_less@4.1.3/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:54405:15)

4:55:24 PM [vite] Error when evaluating SSR module /src/routes/(member)/register/+page.server.ts: failed to import "/node_modules/@messages/prisma/client/index-browser.js"
|- ReferenceError: exports is not defined
    at eval (/Users/my-user/sandbox/sw-society/node_modules/@messages/prisma/client/index-browser.js:4:23)
    at instantiateModule (file:///Users/my-user/sandbox/sw-society/node_modules/.pnpm/vite@4.3.9_@types+node@20.3.1_less@4.1.3/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:54405:15)

Internal server error: exports is not defined
      at eval (/Users/my-user/sandbox/sw-society/node_modules/@messages/prisma/client/index-browser.js:4:23)
      at instantiateModule (file:///Users/my-user/sandbox/sw-society/node_modules/.pnpm/vite@4.3.9_@types+node@20.3.1_less@4.1.3/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:54405:15)
ReferenceError: exports is not defined
    at eval (/Users/my-user/sandbox/sw-society/node_modules/@messages/prisma/client/index-browser.js:4:23)
    at instantiateModule (file:///Users/my-user/sandbox/sw-society/node_modules/.pnpm/vite@4.3.9_@types+node@20.3.1_less@4.1.3/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:54405:15)
```

### Expected behavior

No error, page loading normally

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
  output   = "../node_modules/@messages/prisma/client"
}

datasource db {
  provider = "mongodb"
  url      = env("MESSAGE_DATABASE_URL")
}

type MessagesComments {
  comment     String
  commentator String
}

model messages {
  id       String             @id @default(auto()) @map("_id") @db.ObjectId
  author   String
  comments MessagesComments[]
  post     String
}
```

```ts
import { PrismaClient } from '../../node_modules/@messages/prisma/client';
```

### Environment & setup

- OS: macOS
- Database: 1: PostgreSQL, 2: MongoDB
- Node.js version: 18.13.0
- Package Manager: `pnpm`

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 4.15.0
@prisma/client          : 4.15.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 8fbc245156db7124f997f4cecdd8d1219e360944 (at node_modules/.pnpm/@prisma+engines@4.15.0/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 8fbc245156db7124f997f4cecdd8d1219e360944 (at node_modules/.pnpm/@prisma+engines@4.15.0/node_modules/@prisma/engines/migration-engine-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.15.0-28.8fbc245156db7124f997f4cecdd8d1219e360944
Default Engines Hash    : 8fbc245156db7124f997f4cecdd8d1219e360944
Studio                  : 0.484.0
```

**Comment by ludralph** [comment]
Hi @xenogew 👋 

Have you tried any of the suggested solutions in this [Stackoverflow thread](https://stackoverflow.com/questions/43042889/typescript-referenceerror-exports-is-not-defined)?

**Comment by xenogew** [comment]
Hello @ludralph 
I tried all of answers in that thread, but it's not work for me. 🥲

**Reply by ludralph** [reply]
Hi @xenogew, can you replace this line 

```js
import { PrismaClient } from '../../node_modules/@messages/prisma/client';
```
 with 
```js
import { PrismaClient } from "@messages/prisma/client";
```
from your step 5.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by xenogew** [reply]
Hi @ludralph 
I've tried it both, before I created the question post here.
But yes, it came back still `exports` not defined.

**Reply by ludralph** [reply]
Hi @xenogew, can you please share your repository so I can investigate it further? I was able to reproduce the issue but my last suggestion fixed it for me. Please share a reproducible repository

**Reply by xenogew** [reply]
Hello @ludralph sure,
this is my repository.
https://github.com/xenogew/sw-society

**Reply by ludralph** [reply]
Hi @xenogew, the repository you shared doesn't have any Prisma schema. Are you sharing the correct one?

**Reply by xenogew** [reply]
> Hi @xenogew, the repository you shared doesn't have any Prisma schema. Are you sharing the correct one?

Hi @ludralph 
My apologies, I forgot to tell you my updated code is on the `develop` branch.
Could you checkout from it?

**Reply by ludralph** [reply, marked as answer]
Hi @xenogew, I was able to reproduce your issue on the `develop` branch. These are the steps I took to resolve it:

1. I updated the `schema.prisma` file in the `prisma-message` folder to look like the code below:
```prisma
generator client {
  provider = "prisma-client-js"
  output = "../node_modules/@prisma-message/client"
}

datasource db {
  provider = "mongodb"
  url      = env("MESSAGE_DATABASE_URL")
}

type MessagesComments {
  comment     String
  commentator String
}

model messages {
  id       String             @id @default(auto()) @map("_id") @db.ObjectId
  author   String
  comments MessagesComments[]
  post     String
}
```
2. I added the code below to the `database.ts` file in the `lib` folder
```js
import { PrismaClient as MemberPrisma } from '@prisma/client';
import { PrismaClient } from '@prisma-message/client';

export const dbMember = new MemberPrisma();
export const dbMessage = new PrismaClient();
```
3. In `package.json` file, I add a `mongo:generate` script to generate the Prisma Client for the second schema. You only did the default `npx prisma generate` for the first schema. You also need to generate the Prisma Client for the second schema.

```json
"mongo:generate": "npx prisma generate --schema prisma-message/schema.prisma",
```
4. I added the lines below to the tsconfig file as well
```json
  "module": "es2020",
  "target": "ES2020"
```
Then I ran `pnpm run dev` to compile and run the application. Do let me know if you have any issues trying the above steps.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by xenogew** [reply]
Hello @ludralph 
Yes, it's fixed in the end, thanks for you investigation and solutions.

But if I would like to ask you for your time once more,
if you could give me in details what's the root cause of this issue?

I'd like to understand more in detail of how to think and investigate to the cause of problem. 🙏🏼
(my apologies, but I'm a Typescript beginner, this project is my learning purpose).

**Reply by ludralph** [reply]
Hi @xenogew, I'm glad this issue is fixed now. The root cause is that you didn't generate the `Prisma Client` for the second schema, so the `import` you used in the `database.ts` file could not be resolved accordingly. The key takeaway is always to ensure your Prisma Client is generated. I hope this helps.

**Reply by xenogew** [reply]
Hi @ludralph 
Um, if in that case, I'm sure I followed the blog as instruction and they mentioned to me to generate the file of the second schema.
The generated will be in `node_modules`, so it was ignored within `.gitignore`.
But your answer gave me something, the point is naming and path referencing issue, I guessed.

What I guessed is...
My second schema in `prisma-message` folder, but the `output` I specified with `../node_modules/@message/prisma/client` , but after I change `output` to matched the exact folder name `../node_modules/@prisma-message/client` the issue is solved. 

Thanks for your time and instructions, it helped me lot. 🙏🏼

**Reply by ludralph** [reply]
You're welcome 🙏

**Comment by edmondop** [comment]
@ludralph the problem seems that Sveltekit handle differently what you store in `node_modules`. If you put the generated code let's say in src/generated, this will fail. Any idea why that happens?
